src/wp-admin/network/themes.php

```diff
@@ -208,6 +208,94 @@
+
+			$auto_updates[] = $_GET['theme'];
+			$auto_updates   = array_unique( $auto_updates );
+			// Remove themes that have been deleted since the site option was last updated.
+			$auto_updates = array_intersect( $auto_updates, array_keys( $all_items ) );
+
+			update_site_option( 'auto_update_themes', $auto_updates );
+
+			wp_safe_redirect( add_query_arg( 'enabled-auto-update', 1, $referer ) );
+			exit;
+		case 'disable-auto-update':
+			if ( ! ( current_user_can( 'update_themes' ) && wp_is_auto_update_enabled_for_type( 'theme' ) ) ) {
+				wp_die( __( 'Sorry, you are not allowed to disable themes automatic updates.' ) );
+			}
+
+			check_admin_referer( 'updates' );
+
+			$all_items    = wp_get_themes();
+			$auto_updates = (array) get_site_option( 'auto_update_themes', array() );
+
+			$auto_updates = array_diff( $auto_updates, array( $_GET['theme'] ) );
+			// Remove themes that have been deleted since the site option was last updated.
+			$auto_updates = array_intersect( $auto_updates, array_keys( $all_items ) );
+
+			update_site_option( 'auto_update_themes', $auto_updates );
+
+			wp_safe_redirect( add_query_arg( 'disabled-auto-update', 1, $referer ) );
+			exit;
+		case 'enable-auto-update-selected':
+			if ( ! ( current_user_can( 'update_themes' ) && wp_is_auto_update_enabled_for_type( 'theme' ) ) ) {
+				wp_die( __( 'Sorry, you are not allowed to enable themes automatic updates.' ) );
+			}
+
+			check_admin_referer( 'bulk-themes' );
+
+			$themes = isset( $_POST['checked'] ) ? (array) wp_unslash( $_POST['checked'] ) : array();
+
+			$all_items    = wp_get_themes();
+			$auto_updates = (array) get_site_option( 'auto_update_themes', array() );
+
+			$new_auto_updates = array_merge( $auto_updates, $themes );
+			$new_auto_updates = array_unique( $new_auto_updates );
+			// Remove themes that have been deleted since the site option was last updated.
+			$new_auto_updates = array_intersect( $new_auto_updates, array_keys( $all_items ) );
+
+			update_site_option( 'auto_update_themes', $new_auto_updates );
+
+			wp_safe_redirect( add_query_arg( 'enabled-auto-update', count( $themes ), $referer ) );
+			exit;
+		case 'disable-auto-update-selected':
+			if ( ! ( current_user_can( 'update_themes' ) && wp_is_auto_update_enabled_for_type( 'theme' ) ) ) {
+				wp_die( __( 'Sorry, you are not allowed to disable themes automatic updates.' ) );
+			}
+
+			check_admin_referer( 'bulk-themes' );
+
+			$themes = isset( $_POST['checked'] ) ? (array) wp_unslash( $_POST['checked'] ) : array();
+
+			$all_items    = wp_get_themes();
+			$auto_updates = (array) get_site_option( 'auto_update_themes', array() );
+
+			$new_auto_updates = array_diff( $auto_updates, $themes );
+
+			if ( $new_auto_updates == $auto_updates ) {
+				if ( false === strpos( $referer, '/network/themes.php' ) ) {
+					wp_redirect( network_admin_url( 'themes.php?disabled-auto-update=1' ) );
+				} else {
+					wp_safe_redirect( $referer );
+				}
```

Comment:
Wondering if overriding of the `$referrer` is needed here, and if yes, wouldn't it be needed above too?

Also, this handles cases where there is nothing to save (`$themes` is empty or doesn't contain valid data). Should there be an error message there or just silent fail?